pkg/dogstatsd/server.go

```diff
@@ -92,7 +95,27 @@ func (s *Server) handleMessages(metricOut chan<- *metrics.MetricSample, eventOut
 		go l.Listen()
 	}
 
+	// Run min(2, NumCPU-2) workers, we dedicate a core to the
+	// listener goroutine and another to aggregator + forwarder
+	workers := runtime.NumCPU() - 2
```

Comment:
nit: I think querying `runtime.GOMAXPROCS()` may make a bit more sense here (it should be set to `runtime.NumCPU()` unless there was an override or such)